dulwich/porcelain.py

```diff
@@ -387,7 +388,7 @@ def remove(repo=".", paths=None, cached=False):
         index = r.open_index()
         for p in paths:
             full_path = os.path.abspath(p).encode(sys.getfilesystemencoding())
-            tree_path = path_to_tree_path(r.path, p)
+            tree_path = path_to_tree_path(p)
```

Comment:
tree_path needs to be relative to the repository path; where is this enforced now? os.getcwd() is not necessarily r.path.

Reply:
Well it was never enforced here apparently, as you can see from the diff path_to_tree_path did not enforce it before since the os.path.relpath did not change the value of path

Reply:
anyway since path is either going to be a relative path to cwd or an absolute path we could just systematically check in `path_to_tree_path` with `isabs` and apply `relpath` if False

Reply:
Agreed. IMO that was the bug, not that r.path was passed into path_to_tree_path in the first place.